src/main/java/mel/main/Mel.java

```diff
@@ -0,0 +1,70 @@
+package mel.main;
+
+import mel.exceptions.MelException;
+import mel.exceptions.TaskException;
+import mel.tasks.TaskList;
+import mel.utils.Storage;
+import mel.utils.UI;
+
+import java.util.Scanner;
+
+/**
+ * Mel class is the main class of Mel chatbot.
+ */
+public class Mel {
+    private final TaskList taskList;
+    private final UI ui;
+
+    /**
+     * Constructor for Mel chatbot, instantiates
+     * TaskList, Storage and UI elements.
+     */
+    public Mel() {
+        Storage storage = new Storage();
+        taskList = new TaskList(this, storage);
+        ui = new UI(this);
+    }
+
+    /**
+     * Passes task input to TaskList.
+     * @param input task input string.
+     * @see TaskList
+     */
+    public void taskAction(String input) {
```

Comment:
Could use a better name here? How about something like executeTask? Function names should be verbs.